src/osp/Active/SysNewton.h

```diff
@@ -193,77 +197,51 @@ class SysNewton : public IDynamicSystem
     */
-    static Magnum::Vector4 compute_body_CoM(ActiveScene& rScene,
-        ActiveEnt root, Matrix4 currentTransform = Matrix4());
-
-    /**
-     * Query Newton for the center of mass of the specified rigidbody
-     * 
-     * @param body [in] ACompNwtBody containing a pointer to a NewtonBody
-     * 
-     * @return a 3-vector representing the CoM offset from the body origin
-     */
-    static Vector3 get_rigidbody_CoM(ACompNwtBody const& body);
-
-    /**
-     * Compute the volume of a part
-     *
-     * Traverses the immediate children of the specified entity and sums the
-     * volumes of any detected collision volumes. Cannot account for overlapping
-     * collider volumes.
-     * 
-     * @param rScene [in] ActiveScene containing relevant scene data
-     * @param part   [in] The part
-     * 
-     * @return The part's collider volume
-     */
-    static float compute_part_volume(ActiveScene& rScene, ActiveEnt part);
-
-    /**
-     * Compute the moment of inertia of a massive entity
-     *
-     * Searches the immediate children of the specified entity and computes
-     * its moment of inertia. The entity must have child nodes containing
-     * ACompCollisionShape components to compute its physical extent.
-     * Assumes mass is evenly distributed over collision volume and that the
-     * part's center of mass lies at its root origin.
-     * 
-     * @param rScene [in] ActiveScene containing relevant scene data
-     * @param part   [in] The entity with an ACompMass
-     * 
-     * @return The inertia tensor of the part about its origin
-     */
-    static Matrix3 compute_mass_inertia(ActiveScene& rScene, ActiveEnt part);
+    static Magnum::Vector4 compute_hier_CoM(ActiveScene& rScene,
+        ActiveEnt root, bool includeRootMass=false);
 
     /**
      * Compute the moment of inertia of a rigid body
      *
      * Searches the child nodes of the root and computes the total moment of
-     * inertia of the body. Does not perform recursion, as vehicles do not yet
-     * have nested hierarchies.
-     * 
+     * inertia of the body. To contribute to the inertia of the rigidbody, child
+     * entities must posses both an ACompMass and ACompShape component, so that
+     * the mass distribution of the entity may be calculated.
+     *
      * @param rScene       [in] ActiveScene containing relevant scene data
      * @param root         [in] The root entity of the rigid body
-     * @param centerOfMass [in] The center of mass of the rigid body
-     * 
-     * @return The inertia tensor of the rigid body about its center of mass
+     *
+     * @return The inertia tensor of the rigid body about its center of mass, and 
+     *         a 4-vector containing xyz=CoM, w=total mass
      */
-    static Matrix3 compute_body_inertia(ActiveScene& rScene, ActiveEnt root,
-        Vector3 centerOfMass, Matrix4 currentTransform);
+    static std::pair<Matrix3, Magnum::Vector4> compute_hier_inertia(ActiveScene& rScene,
```

Comment:
Inconsistant namespacing? Should Magnum:: be added to Matrix3? or removed from Vector4?

Reply:
The most common Magnum types are `using`'d in one of the common osp files, maybe Vector4 should be included there.

Reply:
My recommendation is to use Magnum:: everywhere :-)

But consistency is nice, one way or another.